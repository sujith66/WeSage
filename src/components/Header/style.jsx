import styled from "styled-components";

export const HeaderContainer = styled("header")(() => ({
  ".MuiPaper-root": {
    backgroundColor: "transparent",
  },
  ".MuiTypography-h6": {
    fontWeight: "bolder",
    fontSize: "30px",
    color: "white",
    fontStyle: "italic",
    marginRight: "30rem",
  },
  nav: {
    background: "white",
    borderRadius: "7px",
    position: 'absolute',
    right: 0,
    "ul,li": {
      display: "flex",
    },
    li: {
      borderRight: "1px solid rgba(23,24,26,.11)",
    },
    a: {
      textDecoration: "none",
    },
  },
}));
