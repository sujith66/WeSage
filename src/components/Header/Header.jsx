import React from "react";
import Typography from "@mui/material/Typography";
import Link from "@mui/material/Link";
import Toolbar from "@mui/material/Toolbar";
import { HeaderContainer } from "./style";
import Container from "@mui/material/Container";

const Header = () => {
  return (
    <Container>
      {/* container for handling header component */}
      <HeaderContainer
        position="static"
        elevation={0}
        sx={{ borderBottom: (theme) => `1px solid ${theme.palette.divider}` }}
      >
        <Toolbar sx={{ flexWrap: "wrap" }}>
          <Typography variant="h6" color="inherit" noWrap sx={{ flexGrow: 1 }}>
            we
          </Typography>
          <nav>
            <ul>
              <li>
                <Link
                  variant="button"
                  color="text.primary"
                  href="#"
                  sx={{ my: 1, mx: 1.5 }}
                >
                  Help
                </Link>
              </li>

              <li>
                <Link
                  variant="button"
                  color="text.primary"
                  href="#"
                  sx={{ my: 1, mx: 1.5 }}
                >
                  Company
                </Link>
              </li>

              <li>
                <Link
                  variant="button"
                  color="text.primary"
                  href="#"
                  sx={{ my: 1, mx: 1.5 }}
                >
                  Pricing
                </Link>{" "}
              </li>

              <li>
                <Link
                  variant="button"
                  color="text.primary"
                  href="#"
                  sx={{ my: 1, mx: 1.5 }}
                >
                  Sign up
                </Link>{" "}
              </li>

              <li>
                <Link
                  variant="button"
                  color="text.primary"
                  href="#"
                  sx={{ my: 1, mx: 1.5 }}
                >
                  Log in
                </Link>
              </li>
            </ul>
          </nav>
        </Toolbar>
      </HeaderContainer>
    </Container>
  );
};

export default Header;
