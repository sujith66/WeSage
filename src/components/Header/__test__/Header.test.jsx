import React from "react";
import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import Header from "../Header";

const mockHeader = ()=>(    <Header />);


//Test suite which will contain all the tests for Header
describe("Testing Header", () => {
  test("check if the  Header component is rendered successfully", () => {
    render(mockHeader());
    expect(screen.getByText("Help")).toBeInTheDocument();
  });

  test("matches the snapshot to check if the component renders correctly", () => {
    const tree = renderer
      .create((mockHeader()))
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});