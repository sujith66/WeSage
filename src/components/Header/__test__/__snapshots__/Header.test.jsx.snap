// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Header matches the snapshot to check if the component renders correctly 1`] = `
<div
  className="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
>
  <header
    className="sc-bdvvtL fmlPYq"
    elevation={0}
  >
    <div
      className="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1t018ql-MuiToolbar-root"
    >
      <h6
        className="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap css-uh02xs-MuiTypography-root"
      >
        we
      </h6>
      <nav>
        <ul>
          <li>
            <a
              className="MuiTypography-root MuiTypography-button MuiLink-root MuiLink-underlineAlways css-1xkhg5k-MuiTypography-root-MuiLink-root"
              href="#"
              onBlur={[Function]}
              onFocus={[Function]}
            >
              Help
            </a>
          </li>
          <li>
            <a
              className="MuiTypography-root MuiTypography-button MuiLink-root MuiLink-underlineAlways css-1xkhg5k-MuiTypography-root-MuiLink-root"
              href="#"
              onBlur={[Function]}
              onFocus={[Function]}
            >
              Company
            </a>
          </li>
          <li>
            <a
              className="MuiTypography-root MuiTypography-button MuiLink-root MuiLink-underlineAlways css-1xkhg5k-MuiTypography-root-MuiLink-root"
              href="#"
              onBlur={[Function]}
              onFocus={[Function]}
            >
              Pricing
            </a>
             
          </li>
          <li>
            <a
              className="MuiTypography-root MuiTypography-button MuiLink-root MuiLink-underlineAlways css-1xkhg5k-MuiTypography-root-MuiLink-root"
              href="#"
              onBlur={[Function]}
              onFocus={[Function]}
            >
              Sign up
            </a>
             
          </li>
          <li>
            <a
              className="MuiTypography-root MuiTypography-button MuiLink-root MuiLink-underlineAlways css-1xkhg5k-MuiTypography-root-MuiLink-root"
              href="#"
              onBlur={[Function]}
              onFocus={[Function]}
            >
              Log in
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </header>
</div>
`;
